broker-core/src/main/java/io/zeebe/broker/clustering/ClusterComponent.java

```diff
@@ -106,29 +103,15 @@ private void initGossip(
       final CompositeServiceBuilder baseLayerInstall,
       final SystemContext context,
       final NodeInfo localMember) {
-    final GossipService gossipService = new GossipService(context.getBrokerConfiguration());
-    baseLayerInstall
-        .createService(GOSSIP_SERVICE, gossipService)
-        .dependency(
-            TransportServiceNames.clientTransport(TransportServiceNames.MANAGEMENT_API_CLIENT_NAME),
-            gossipService.getClientTransportInjector())
-        .dependency(
-            TransportServiceNames.bufferingServerTransport(
-                TransportServiceNames.MANAGEMENT_API_SERVER_NAME),
-            gossipService.getBufferingServerTransportInjector())
-        .install();
-
-    // TODO: decide whether failure to join gossip cluster should result in broker startup fail
-    final GossipJoinService gossipJoinService =
-        new GossipJoinService(context.getBrokerConfiguration().getCluster(), localMember);
-    baseLayerInstall
-        .createService(GOSSIP_JOIN_SERVICE, gossipJoinService)
-        .dependency(GOSSIP_SERVICE, gossipJoinService.getGossipInjector())
-        .install();
 
     final AtomixService atomixService = new AtomixService(context.getBrokerConfiguration());
-
     baseLayerInstall.createService(ATOMIX_SERVICE, atomixService).install();
+
+    final AtomixJoinService atomixJoinService = new AtomixJoinService();
+    baseLayerInstall
+        .createService(ATOMIX_JOIN_SERVICE, atomixJoinService)
+        .dependency(ATOMIX_SERVICE, atomixJoinService.getAtomixInjector())
```

Comment:
Can we have also a dependency on `TOPOLOGY_MANAGER_SERVICE`, so we can be sure that they start in order. I think this is also a bug in the old implementation.